### PR TITLE
docs: update docs

### PR DESCRIPTION
**Pull request by hallazzang**
## Description

Additionally update docs.

Also fix a minor issue mentioned in https://github.com/tendermint/farming/pull/175/files#r730641621.

### CODE REVIEW

File: x/farming/spec/03_state_transitions.md [hunk 1]
```diff
@@ -49,15 +49,15 @@ When a farmer stakes an amount of coins, the following state transitions occur:
...
+- it imposes more gas if the farmer already has `Staking` with the same coin denom(see [07_params.md](07_params.md#DelayedStakingGasFee) for details)
 
 ## Unstake
 
 When a farmer unstakes an amount of coins, the following state transitions occur:
 
 - it adds `Staking` and `QueueStaking` amounts to see if the unstaking amount is sufficient
 - it automatically withdraws rewards for the coin denom which are accumulated over the last epochs
-- it substracts the unstaking amount of coins from `QueueStaking` first and if it is not sufficient then it substracts from `Staking`
+- it subtracts the unstaking amount of coins from `QueueStaking` first and if it is not sufficient then it substracts from `Staking`
```
Comment (jaybxyz) [inline x/farming/spec/03_state_transitions.md]:
```suggestion
- it subtracts the unstaking amount of coins from `QueueStaking` first and if it is not sufficient then it subtracts from `Staking`
```

File: x/farming/spec/03_state_transitions.md [hunk 2]
```diff
@@ -24,7 +24,7 @@ const (
 
 - Staking Coins for Farming
   - Each `farmingPlan` predefines list of `stakingCoinWeights` using `sdk.DecCoins`
-  - `weight` mean that each group of stakers with each coin `denom` will receive each predefined `weight` of the total rewards
+  - `weight` mean that each group of farmers with each coin `denom` will receive each predefined `weight` of the total rewards
```
Comment (jaybxyz) [inline x/farming/spec/03_state_transitions.md]: It's great that you work on this. I didn't revise this part in #172. This sentence is actually awkward to me. Let me revise this.

**Review by dongsam — APPROVED** [review]

File: x/farming/spec/03_state_transitions.md [hunk 3]
```diff
@@ -24,7 +24,7 @@ const (
 
 - Staking Coins for Farming
   - Each `farmingPlan` predefines list of `stakingCoinWeights` using `sdk.DecCoins`
```
Comment (jaybxyz) [inline x/farming/spec/03_state_transitions.md]:
```suggestion
  - Each `Plan` defines a list of `StakingCoinWeights` using `sdk.DecCoins`
```

File: x/farming/spec/03_state_transitions.md (hunk 2 above)
Comment (jaybxyz) [inline x/farming/spec/03_state_transitions.md]:
```suggestion
  - Each weight in `StakingCoinWeights` is calculated in accordance with the total rewards and farmers who stake the coin denom defined in `StakingCoinWeights` will receive the relative amount of rewards.
```

**Review by jaybxyz — CHANGES_REQUESTED** [review]

File: x/farming/spec/03_state_transitions.md (hunk 2 above)
Comment (hallazzang) [reply x/farming/spec/03_state_transitions.md]: OK!

**Review by jaybxyz — APPROVED** [review]

**Pull request merged**